### PR TITLE
Add optional parameter to configure start method for multiprocessing

### PR DESCRIPTION
**Pull request by philippotto**
See https://github.com/scalableminds/cluster_tools/pull/87 for a bit of context. Callers of `get_executor` may now configure the proper spawn method. When we drop python 3.6 support, we can construct a proper context with the appropriate `start_method` instead of setting that value globally.

### CODE REVIEW

**Review by georgwiese — APPROVED** [review]
Awesome!

File: cluster_tools/__init__.py [hunk 1]
```diff
@@ -24,12 +25,26 @@ class WrappedProcessPoolExecutor(ProcessPoolExecutor):
     def __init__(self, **kwargs):
         new_kwargs = get_existent_kwargs_subset(PROCESS_POOL_KWARGS_WHITELIST, kwargs)
 
-        # For the purpose of these cluster tools, it shouldn't make a significant difference
-        # whether we use spawn or fork. However, TensorFlow is not fork-safe, see:
-        # https://github.com/tensorflow/tensorflow/issues/5448#issuecomment-258934405
-        multiprocessing.set_start_method("spawn", force=True)
+        self.did_overwrite_start_method = False
+        if "start_method" in kwargs is not None:
```
Comment (georgwiese) [inline cluster_tools/__init__.py:29]:
```suggestion
        if "start_method" in kwargs and kwargs["start_method"] is not None:
```

**Review by georgwiese — CHANGES_REQUESTED** [review]
I think I found a bug.

File: cluster_tools/__init__.py (hunk 1 above)
Comment (georgwiese) [reply cluster_tools/__init__.py:29]: Maybe you can add a test also for the start_method=None case, as this is how we'd use it in voxelytics.

Comment (philippotto) [reply cluster_tools/__init__.py]: > Maybe you can add a test also for the start_method=None case, as this is how we'd use it in voxelytics.

I added that test now. However, it passed before this fix, since set_start_method was fine with being passed None. However, it's obviously cleaner to completely ignore the parameter if its value is None. Thank you for spotting this!

**Review by georgwiese — APPROVED** [review]
:heart:
